ledger/src/advance.rs

```diff
@@ -104,13 +104,14 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
                 "The given block is not the direct successor of the latest block"
             );
         } else {
+            ensure!(block.height() > 0, "Non-genesis blocks cannot have height 0");
```

Comment:
```suggestion
            ensure!(block.height() != 0, "Non-genesis blocks cannot have height 0");
```
nit: some day, `clippy` might complain about this

Reply:
Good catch!